tests/rust/assoc_types.rs

```diff
@@ -0,0 +1,90 @@
+pub trait MyTrait{
+    type SomeType;
+}
+
+pub trait AnotherTrait{
+    type AnotherType;
+}
+
+#[repr(C)]
+pub struct MyStruct{
+    a: u8,
+}
+
+impl MyTrait for MyStruct{
+    type SomeType = i64;
+}
+
+impl AnotherTrait for MyStruct{
+    type AnotherType = [u8; 5];
+}
+
+impl MyTrait for i32{
+    type SomeType = bool;
+}
+
+impl AnotherTrait for i32{
+    type AnotherType = u64;
+}
+
```

Comment:
I want you to write a test that covers an example like this:

```rs
trait BlanketBound {}

impl BlanketBound for i32 {}

trait Blanket {
    type Inner;
}

impl<T: BlanketBound> Blanket for T {
    type Inner = u64;
}

fn test_fn(arg: <i32 as BlanketBound>::Inner) {}
```

Reply:
you can put this test in a separate file or not, as you see fit

Reply:
when you make this test work, you are finished with this PR. Then we can open a PR into `cbindgen`